packages/types/src/interfaces/integriteeWorker/definitions.ts

```diff
@@ -190,9 +190,26 @@ export default {
     },
     TimestampSetArgs: '(AccountId, H160, BalanceType)',
     BalanceTransferArgs: '(AccountId, AccountId, BalanceType)',
+    BalanceTransferWithNoteArgs: '(AccountId, AccountId, BalanceType, String)',
     BalanceShieldArgs: '(AccountId, AccountId, BalanceType, ParentchainId)',
     BalanceUnshieldArgs: '(AccountId, AccountId, BalanceType, ShardIdentifier)',
-    BalanceSetBalanceArgs: '(AccountId, AccountId, BalanceType, BalanceType)',
+    NotesForArgs: '(AccountId, BucketIndex)',
+    BucketIndex: 'u32',
+    NoteIndex: 'u64',
+    TrustedNote: {
+      _enum: {
+        TrustedCall: 'Vec<u8>',
+        SgxRuntimeEvent: 'Vec<u8>'
+      }
+    },
+    BucketInfo: {
+      index: 'BucketIndex',
+      bytes: 'u32',
+    },
+    NotesBucketInfo: {
+      first: 'BucketInfo',
+      last: 'BucketInfo',
+    },
```

Comment:
Note, the WIP in the worker uses a tuple here, but I decided it is more convenient to use a struct, and the worker will change this hopefully soon. However, we can already use a struct here because the scale codec is compatible.